File: c_day3/program 3.c
/******************************************************************************

                            Online C Compiler.
                Code, Compile, Run and Debug C program online.
Write your code in this editor and press "Run" button to compile and execute it.

*******************************************************************************/

#include <stdio.h>
#include <stdbool.h>
#include<math.h>

int main()
{
   int a[100];
   int n;
    int sum=0;
   int max1;
   printf("enter the size of the array:");
   scanf("%d",&n);
   
   for(int i=0;i<n;i++)
   {
       scanf("%d",&a[i]);
   }
   max1=a[0];
   for(int i=0;i<n;i++)
   {
      sum=sum+a[i];
      
      if(sum>max1)
      {
          max1=sum;
      }
      
      if(sum<0)
      {
          sum=0;
      }
       
   }
   printf("maximum sum is %d",max1);
}

File: java_day1/program 3.java
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import java.util.* ;
class HelloWorld {
    public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.print("Enter a  number :");
    int n =sc.nextInt();
   
        int slow = n, fast = n;

        do {
            slow = sum(slow);
            fast = sum(sum(fast));
        } while (slow != fast);

        System.out.println(slow == 1);  
    }

    static int sum( int n) {
        int sum = 0;
        while (n > 0) {
            int digit = n % 10;
            sum += digit * digit;
            n /= 10;
        }
        return sum;
   
    }
}